include/deal.II/grid/grid_generator.h

```diff
@@ -1575,6 +1575,9 @@ namespace GridGenerator
    * subset of faces of volume cells for which this function should extract
    * the surface mesh. If left at its default, i.e., if the set is empty, then
    * the function operates on <i>all</i> boundary faces.
+   * @param[in] cell_orientation A boolean determining whether the normal
```

Comment:
What is the motivation for this parameter? Wouldn't it be acceptable to always have outward facing normals?

Reply:
Second, I would call this parameter ``make_normal_vectors_point_outwards`` or something like that.

Reply:
The motivation is to retain as much backward compatibility (until now, the normal vector pointed inward in 3d; and either inward or outward in 2d, depending on the situation) as possible while also allowing to choose the often used convention of outward pointing normals. Personally, I really don't care whether the normal vector points inward or outward. So I could on the one hand also imagine to omit the new parameter and let normals always point inwards. On the other hand, I think it's not a big deal to introduce this new parameter and let the user choose.

Reply:
I think tagging additional ``bool`` parameters with default values to the end of a function is a sign for a bad design. I think always using a defined direction is fixing a bug. I would have expected to always get outward facing normals.

Reply:
I appreciate the concern for backwards compatibility but I agree with @tjhei: I cannot come up with a situation where we would not want outward-facing normal vectors.

If anyone ever needs this we could simply add `GridTools::invert_normal_orientation` or something like that instead of bundling it with the mesh extraction function.

Reply:
I interpreted Wolfgang's answer [here](https://groups.google.com/d/msg/dealii/AzHUdaDaifI/q8BJPTCCAgAJ) as that it would be good to stay backward compatible. If not so, I absolutely agree that it's best to eliminate the bool and let the normals always point outward.

Reply:
I will mark this as resolved since we have agreed below to have outward normals.